test/test_main.py

```diff
@@ -132,10 +132,10 @@ def _simulate_event(self, event_spec):
                 'JUJU_RELATION_ID': str(event_spec.relation_id),
             })
             remote_app = event_spec.remote_app
-            if remote_app is None:
-                remote_app = ''
-
-            env['JUJU_REMOTE_APP'] = remote_app
+            # A remote app may not be in an event spec and so JUJU_REMOTE_APP will not be populated
+            # but it may still be a relation event with JUJU_REMOTE_UNIT for Juju < 2.7.
+            if remote_app is not None:
+                env['JUJU_REMOTE_APP'] = remote_app
```

Comment:
The first sentence is the code stated with English, which is not useful. Let's please just drop the whole comment and have a line right above the env line with:
```
# For juju < 2.7 app name is extracted from JUJU_REMOTE_UNIT.
```

The second sentence is completely unrelated to the code here. It's the test that verifies such conditions, not the simulation logic. I suggest keeping the code and dropping the comment.